android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -123,21 +126,53 @@ public class RNSentryModuleImpl {
     private String cacheDirPath = null;
     private ISentryExecutorService executorService = null;
 
+    private final @NotNull Runnable emitNewFrameEvent;
+
     /** Max trace file size in bytes. */
     private long maxTraceFileSize = 5 * 1024 * 1024;
 
     public RNSentryModuleImpl(ReactApplicationContext reactApplicationContext) {
-        packageInfo = getPackageInfo(reactApplicationContext);
-        this.reactApplicationContext = reactApplicationContext;
+      packageInfo = getPackageInfo(reactApplicationContext);
+      this.reactApplicationContext = reactApplicationContext;
+      this.emitNewFrameEvent = createEmitNewFrameEvent();
     }
 
     private ReactApplicationContext getReactApplicationContext() {
-        return this.reactApplicationContext;
+      return this.reactApplicationContext;
+    }
+
+    private @Nullable Activity getCurrentActivity() {
+      return this.reactApplicationContext.getCurrentActivity();
+    }
+
+    private @NotNull Runnable createEmitNewFrameEvent() {
+        final @NotNull SentryDateProvider dateProvider = new SentryAndroidDateProvider();
+
+        return () -> {
+          final SentryDate endDate = dateProvider.now();
+          WritableMap event = Arguments.createMap();
+          event.putDouble("newFrameTimestampInSeconds", endDate.nanoTimestamp() / 1e9);
```

Comment:
Not sure if it's fine, but `nanoTimestamp()` will only give you a millisecond precision, [see the issue linked here](https://github.com/getsentry/sentry-java/issues/3210).
I see the duration is calculated via `this._latestTransaction?.startTimestamp`, but couldn't find the setter for `.startTimestamp`.

Reply:
Thank you for the note, the RN measurements have the nanoseconds precision ([RN impl of performance now](https://github.com/facebook/react-native/blob/d09c02f9e2d468e4d0bde51890e312ae7003a3e6/packages/react-native/ReactCommon/cxxreact/JSExecutor.cpp#L28)), this is used for the `startTimestamp`.

But it doesn't seem to be an issue, looking at the test transactions.